lib/permutations.rb

```diff
@@ -1,4 +1,21 @@
 
 def permutations?(string1, string2)
-  raise NotImplementedError, "permutations? not implemented" 
+  return true if string1.length == 0 && string2.length == 0
+  hash = {}
+  false if string1.length != string2.length
+  string1 = string1.split
+  string2 = string2.split
+  count = 0
+  string1.each do |letter|
+    hash[letter] = hash[letter].nil? ? 1 : hash[letter] + 1
+  end
+  until count == hash.length
+    string2.each do |letter|
+      if hash[count] != letter
+      return false
+      else count += 1
+      end
+    end
+  end
```

Comment:
For this loop I suggest looping through string2 and counting down the matching hash element, 

```ruby
if hash[letter]
  hash[letter] -= 1
else
  return false
end
```

Then you can use the hash to see if the two strings have the same number of letters.